pkgs/development/compilers/rust/make-rust-platform.nix

```diff
@@ -1,39 +1,46 @@
 {
   lib,
-  buildPackages,
-  callPackages,
   cargo-auditable,
   config,
-  stdenv,
-  runCommand,
-  generateSplicesForMkScope,
   makeScopeWithSplicing',
+  pkgsBuildBuild,
+  pkgsBuildHost,
+  pkgsBuildTarget,
+  pkgsHostHost,
+  pkgsHostTarget,
+  pkgsTargetTarget,
 }@prev:
 
 {
   rustc,
   cargo,
   cargo-auditable ? prev.cargo-auditable,
-  stdenv ? prev.stdenv,
```

Comment:
Overriding `stdenv` like this doesn't really work in practice. Imo this should be removed at least until we figure out how to override it properly. Firstly because there isn't one stdenv when cross-compiling. And secondly because `stdenv` is coupled too strongly in a few places that I haven't been able to unravel.

I attempted to replace it with a function like `stdenvSeletor = pkgs: pkgs.stdenv` which would allow you to select `stdenv` depending on the context (e.g. `pkgsTargetTarget` vs `pkgs`), but that didn't work very well either.